app/src/main/java/com/github/libretube/ui/dialogs/ColorPickerDialog.kt

```diff
@@ -0,0 +1,169 @@
+package com.github.libretube.ui.dialogs
+
+import android.app.Dialog
+import android.content.Context
+import android.graphics.Color
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.EditText
+import android.widget.SeekBar
+import android.widget.Toast
+import com.github.libretube.R
+
+class ColorPickerDialog(
+    context: Context,
+    private val initialColor: Int,
+    private val onColorSelectedListener: OnColorSelectedListener
+) : Dialog(context), SeekBar.OnSeekBarChangeListener, View.OnClickListener {
```

Comment:
I can't seem to get this to work any suggestions? (I've tried a lot of SO solutions :/)

<img width="879" alt="Screen Shot 2023-06-21 at 12 48 21 PM" src="https://github.com/libre-tube/LibreTube/assets/48372480/dc8e5f14-51ad-48d2-b888-dbd618a2fa6b">

Reply:
And for the tag, you can use `this::class.java.name` instead of "acknowledgements"

Reply:
That did it! Why does that work and not something like dialog.childFragmentManager?

Reply:
Preferences just don't have fragment managers and don't include an inbuilt function to get the activity. The activity is at the top of the view hierarchy (so the most important one), so it basically has the control to display fragments, dialogs, ...
Activity inherits from Context, and since we use a `SettingsActivity` we can just cast the type.